lib/jnpr/junos/facts/get_chassis_cluster_status.py

```diff
@@ -17,20 +31,45 @@ def get_facts(device):
     """
     srx_cluster = None
     srx_cluster_id = None
+    redundancy_group = None
 
     try:
-        rsp = device.rpc.get_chassis_cluster_status(normalize=True,
-                                                    redundancy_group="0")
+        rsp = device.rpc.get_chassis_cluster_status(normalize=True)
```

Comment:
@vnitinv I see a few issues with your suggestion.

1) Because fact gathering is now done on demand, I feel there is little benefit from making the RPC call conditional on the personality. As long as the fact SRX-cluster-related facts aren't accessed, this RPC isn't called.

2) The new fact gathering system differentiates between `False` and `None` return values. A return value of `False` means the fact was successfully gathered and it's value is not true. A value of `None` means the fact could not be gathered definitively. If we don't call this RPC, then we can't definitively gather this particular fact. We would potentially be returning a different values than is returned with the current. (See final point.)

3) Right now, the `personality` fact is some of the most fragile fact gathering code. It makes some assumptions that may not continue to hold true as new product models are introduced. Right now a 3-digit SRX model is assumed to be `'SRX_BRANCH'` and a 4-digit SRX model is assumed to be `'SRX_HIGHEND'`. For example, if a 5-digit SRX model is introduced, then the `personality` will be `None`. While that's obviously something that could be updated in the gathering of the `personality` fact, I'd rather not make the other facts dependent upon this fragile fact.

Based on these points, I recommend we not make the RPC call dependent upon the `personality` fact.